node/pkg/cni/token_watch_test.go

```diff
@@ -0,0 +1,90 @@
+package cni_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/calico/node/pkg/cni"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var _ = Describe("FV tests", func() {
+	const serviceAccountName = "cni-token-watch-test"
+	const namespace = metav1.NamespaceSystem
+
+	It("should create a token successfully", func() {
+		clientset := createKubernetesClient()
+		setupServiceAccount(clientset, serviceAccountName, namespace)
+		defer cleanupServiceAccount(clientset, serviceAccountName, namespace)
+
+		tr := cni.NewTokenRefresher(clientset, namespace, serviceAccountName)
+		tu, err := tr.UpdateToken()
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(tu.Token).NotTo(BeEmpty())
+	})
+
+	It("should create a token successfully and deliver it through the channel", func() {
+		clientset := createKubernetesClient()
+		setupServiceAccount(clientset, serviceAccountName, namespace)
+		defer cleanupServiceAccount(clientset, serviceAccountName, namespace)
+
+		tr := cni.NewTokenRefresher(clientset, namespace, serviceAccountName)
+		tokenChan := tr.TokenChan()
+		go tr.Run()
```

Comment:
I think Run() needs to take a "stop" channel so we can signal to stop the token refresher when the test is over. Otherwise we will leak token refresher goroutines. 

```go
stopCh := make(chan struct{})
go tr.Run(stopCh)
defer close(stopCh)
```

Reply:
I added a function for stopping the `TokenRefresher`, but made the channel private.
Is that fine?

Reply:
Yep, that works too :+1: